File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.java
import java.util.Arrays;
class Solution {
    public String solution(int[] numbers) {
        numbers = Arrays.stream(numbers).boxed().sorted((a,b)-> (b+""+a).compareTo(a+""+b)).mapToInt(Integer::intValue).toArray();
        StringBuilder sb = new StringBuilder();
        for(int num : numbers){
            sb.append(num);
        }
               String answer = sb.toString();
        return answer.charAt(0) == '0' ? "0" : answer;
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>
#define str(x, y) (to_string(x) + to_string(y))
using namespace std;
bool compare(int a, int b){
    return str(a, b) > str(b, a);
}
string solution(vector<int> numbers) {
    string answer = "";
    sort(numbers.begin(), numbers.end(), compare);
    if(numbers[0] == 0) return "0";
    else{
        for(auto i : numbers){
            answer += to_string(i);
        }
    }
    return answer;
}